ws-nextjs-app/pages/[service]/homepages/handleHomepageRoute.ts

```diff
@@ -0,0 +1,119 @@
+import { GetServerSidePropsContext } from 'next';
+import extractHeaders from '#server/utilities/extractHeaders';
+import { HOME_PAGE } from '#app/routes/utils/pageTypes';
+import parseRoute from '#app/routes/utils/parseRoute';
+import nodeLogger from '#lib/logger.node';
+import { OK } from '#app/lib/statusCodes.const';
+import { ROUTING_INFORMATION } from '#app/lib/logger.const';
+import getPathExtension from '#app/utilities/getPathExtension';
+import PageDataParams from '#app/models/types/pageDataParams';
+import handleError from '#app/routes/utils/handleError';
+import { getServerExperiments } from '#server/utilities/experimentHeader';
+import shouldRender from '../articles/shouldRender';
```

Comment:
If we're using this on other pages, I think we should move it out in the utilities folder so that it can be shared.

This `shouldRender` function isn't used on other Next.js pages other than articles, but perhaps we should continue using it here since it is being used on the Express version of Homepages.